cloud-computing/gramine/README.md

```diff
@@ -0,0 +1,77 @@
+## How to run demo with Gramine
+
+1. Start a SPS container:
+```shell
+docker run -d -v /opt/multiple/sessions:/graphene/workplace/sessions -v /opt/secret-prov/certs/:/graphene/workplace/certs -p 8080:8080 -p 4433:4433 -e SPS_USERNAME=admin -e SPS_PASSWORD=admin --name iexec-sps iexechub/iexec-sps:bf3bb00-dev
+```
+
+
+2. Build your app (from `cloud-computing` directory) :
+```shell
+docker build -t tee-gramine-python-hello-world:latest -f gramine/Dockerfile  .
+```
+Please note the `measurement` value.
+
+
+3. To add a session to the SPS, run the following after having filled both env var:
+```shell
+SESSION_ID=<define your custom session id>
+MEASUREMENT=<set previous retrieved measurement>
+
+curl --location --request POST 'localhost:8080/api/session/' \
+--header 'Authorization: Basic YWRtaW46YWRtaW4=' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+  "session": "'${SESSION_ID}'",
+  "enclaves": [
+    {
+      "name": "app",
+      "mrenclave": "'${MEASUREMENT}'",
+      "command": "/apploader.sh",
+      "environment": {
+        "IEXEC_IN": "/iexec_in",
+        "IEXEC_OUT": "/iexec_out",
+        "IEXEC_DATASET_FILENAME": "file.txt",
+        "IEXEC_INPUT_FILES_NUMBER": "1",
+        "IEXEC_INPUT_FILE_NAME_1": "file.txt",
+        "IEXEC_TASK_ID": "TASK_ID",
+        "IEXEC_APP_DEVELOPER_SECRET": "App developer secret",
+        "IEXEC_REQUESTER_SECRET_1": "Requester secret 1",
+        "IEXEC_REQUESTER_SECRET_2": "Requester secret 2",
+        "IEXEC_REQUESTER_SECRET_3": "Requester secret 3"
+      },
+      "volumes": [
+        "/iexec_in",
+        "/iexec_out"
+      ]
+    }
+  ]
+}'
+```
+
+
+4. Run the app:
+```shell
+docker run --device=/dev/sgx/enclave -v /iexec_in:/iexec_in -v /iexec_out:/iexec_out -v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket -v $PWD/encryptedData:/workplace/encryptedData -v /opt/secret-prov/certs/:/graphene/attestation/certs/ --net=host -e session=${SESSION_ID} -e sps=localhost:4433  tee-gramine-python-hello-world:latest
```

Comment:
`/dev/sgx_enclave` instead of `/dev/sgx/enclave` ?
What about `/dev/sgx_provision` which is provided as well with a native support in Linux kernel ?
What is the prerequisite to have `/var/run/aesmd/aesmd.sock` ?

Reply:
I'd say this README should be an example for localhost running. Let's keep it simple!
Docker network will be covered in the script example I will provide later :)

Reply:
I just saw, that devices are mounted on `/dev/sgx/enclave` on Azure.
My second question becomes what of `/dev/sgx/provision` ?